Fixtures

Yurasb opened the pull request:
1. adding fixtures for purge, create and redirect
2. optimizing preconditions
3. added tests for empty links list

Yurasb committed:
1. adding fixture for purge, create and redirect
2. optimizing preconditions
3. added tests for empty links list

Yurasb committed:
Adding XML-schema validation of response body for redirect test

v1-wizard commented on tests/conftest.py:
```diff
@@ -0,0 +1,31 @@
+
+@pytest.fixture()
+def purge_all_links(request):
+    requests.request(
```
requests.delete(..) - looks more cool.

v1-wizard commented on tests/conftest.py:
```diff
@@ -0,0 +1,31 @@
+
+@pytest.fixture()
+def create_shortcut_link(request):
+    create = requests.request(
```
requests.post(...)

v1-wizard commented on tests/test_delete_all_links.py:
```diff
@@ -2,13 +2,9 @@
-        url='http://localhost:8888/shortcut',
-        data='{ "link": "https://github.com/Yurasb/url_shortener_testing"}'
-    )
+    create_shortcut_link
```
Wtf? I think you could delete this "#precondition" section.

Yurasb replied:
@v1-wizard Why? We create some shortcut to be sure that we really delete something with delete request

v1-wizard commented on tests/test_delete_all_links.py:
```diff
@@ -2,13 +2,9 @@
 from requests import request
```
Could you add test that really check that all links is delete from db?

Yurasb replied:
@v1-wizard How about try to redirect by deleted id? I think it will show that shortcut is really deleted

v1-wizard commented on tests/test_list_all_links.py:
```diff
@@ -78,7 +91,9 @@ def test_all_links_wrong_method_body():
+            }
         }
     )
     assert v.validate(response.json())
```
I think you should make this assert more verbose. 
`assert v.validate(response.json()), v.errors`

v1-wizard commented on tests/test_redirect.py:
```diff
@@ -4,107 +4,77 @@
+    purge_all_links
+    create_shortcut_link
     # action
     response = request(
```
You should import this things as:
`import requests`
or
`from requests import request`
and use the same in all code. It's called code style. =)

v1-wizard commented on data/schema.xsd:
```diff
@@ -0,0 +1,968 @@
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
Is this full scheme for check html as xml? I think it's very powerful solution little url_shortener. Maybe use something lite? For example: `https://docs.python.org/3/library/html.parser.html`

Yurasb replied:
@v1-wizard I would prefer to use lxml, but assert not by XSD - which is really overhead - but using XPath for particular element on the page to ensure that we get expected page.

v1-wizard reviewed (changes requested):
If you have some question you could contact with me in slack.

Yurasb committed:
Some refactoring

Yurasb committed:
1. fixing redirect tests
2. simplifying body validation for 500 error cases
3. removing data package
4. removing pre-condition statements

Yurasb commented:
Closing to recreate with all changes
